src/openrct2-ui/windows/Finances.cpp

```diff
@@ -318,7 +318,7 @@ class FinancesWindow final : public Window
                 OnPrepareDrawMarketing();
                 break;
             case WINDOW_FINANCES_PAGE_RESEARCH:
-                OnPrepareDrawResearch();
+                WindowResearchFundingPrepareDraw(this, WIDX_RESEARCH_FUNDING);
```

Comment:
`WindowResearchFundingPrepareDraw` This should be a member function, then call it like:
```suggestion
                PrepareDraw(WIDX_RESEARCH_FUNDING);
```

Reply:
But this is calling code from another window

Reply:
Oh huh, so this the this parameter can be either window, and this tab of the finance window and the research window are pretty much copies? This function assumes that they are the exact same, while we have no way to guarantee this. That's kind of ugly design. :/

Reply:
Yeah, they appear the same, but at least using the same code. I agree it's ugly, but I don't think it should be tackled onto this PR.

PS: note this comment applies to other issues you've raised too

Reply:
I agree, that will be out of scope for this PR. I'll make the other comments as resolved, there are a couple other points that need change though.